doc/sphinx/source/developer_guide/abi.rst

```diff
@@ -49,58 +82,68 @@ Include header and declare an ``abi_event`` as a global ``static`` variable (but
      // do something here
     }
 
+.. warning:: Make sure the callback doesn't take too much time (process the received data elsewhere), as ABI callbacks are not thread safe.
```

Comment:
It is true that ABI callback are not thread safe, however making it short or long to process is not related:
- not thread safe: don't use it for inter-thread communication unless you have implement yourself a proper protection for the shared memory
- the ABI callback is called synchronously to the data send function and thus should be short (and obviously non-blocking) to avoid delaying the execution of the rest of the main loop.